client-device-connection-infinispan/src/test/java/org/eclipse/hono/deviceconnection/infinispan/client/HotrodCacheTest.java

```diff
@@ -152,10 +161,149 @@ void testPutFails(final VertxTestContext ctx) {
+        final org.infinispan.client.hotrod.RemoteCache<Object, Object> grid = givenAConnectedCache();
+        when(grid.getWithMetadata(anyString())).thenThrow(new IllegalStateException());
+        cache.connect()
+                .compose(c -> c.getWithVersion("key"))
+                .setHandler(ctx.failing(t -> {
+                    ctx.verify(() -> {
+                        verify(grid).getWithMetadata("key");
+                        assertThat(t).isInstanceOf(IllegalStateException.class);
+                    });
+                    ctx.completeNow();
+                }));
+    }
+
+    /**
+     * Verifies that a request to get a map of all cache entries with given keys
+     * results in the map value being retrieved from the data grid.
+     *
+     * @param ctx The vert.x text context.
+     */
+    @Test
+    void testGetAllSucceeds(final VertxTestContext ctx) {
+        final org.infinispan.client.hotrod.RemoteCache<Object, Object> grid = givenAConnectedCache();
+        final Map<Object, Object> mapValue = new HashMap<>();
+        when(grid.getAll(anySet())).thenReturn(mapValue);
+        final Set<String> keys = Set.of("key");
+        cache.connect()
+                .compose(c -> c.getAll(keys))
+                .setHandler(ctx.succeeding(v -> {
+                    ctx.verify(() -> {
+                        verify(grid).getAll(keys);
+                        assertThat(v).isEqualTo(mapValue);
+                    });
+                    ctx.completeNow();
+                }));
+    }
+
+    /**
+     * Verifies that a request to get a map of all cache entries with given keys
+     * fails with the root cause for the failure to access the data grid.
+     *
+     * @param ctx The vert.x text context.
+     */
+    @Test
+    void testGetAllFails(final VertxTestContext ctx) {
+        final org.infinispan.client.hotrod.RemoteCache<Object, Object> grid = givenAConnectedCache();
+        when(grid.getAll(anySet())).thenThrow(new IllegalStateException());
+        final Set<String> keys = Set.of("key");
+        cache.connect()
+                .compose(c -> c.getAll(keys))
+                .setHandler(ctx.failing(t -> {
+                    ctx.verify(() -> {
+                        verify(grid).getAll(keys);
+                        assertThat(t).isInstanceOf(IllegalStateException.class);
+                    });
+                    ctx.completeNow();
+                }));
+    }
+
+    private org.infinispan.client.hotrod.RemoteCache<Object, Object> givenAConnectedCache() {
+        final Configuration configuration = mock(Configuration.class);
         @SuppressWarnings("unchecked")
-        final BasicCache<Object, Object> result = mock(BasicCache.class);
-        when(remoteCacheManager.getCache(anyString())).thenReturn(result);
+        final org.infinispan.client.hotrod.RemoteCache<Object, Object> result = mock(org.infinispan.client.hotrod.RemoteCache.class);
```

Comment:
Since the `HotrodCache` class uses a `org.infinispan.client.hotrod.RemoteCache` field now (instead of `org.infinispan.commons.api.BasicCache`), this mock had to be changed.
Unfortunately, the `org.infinispan.client.hotrod.RemoteCache` has a dependency on the `infinispan-query-dsl` module (for the `retrieveEntriesByQuery` method) and without having that module integrated, creating the mock here will fail.
Therefore I've added `infinispan-query-dsl` as a test-only dependency (CQ still needed for that). As alternative option, I only see removing this test class.